### PR TITLE
Implement separate needs admin code review and needs admin content review flags

### LINKED ISSUE #6826: Use different admin flag for content review
This is a followup for #6310.

The original issue requested to use the same admin review flag for code review and content review. Discussing with @wagnerand, clearly there are use cases where those flags should work independently. So, we'll need these changes:

- [ ] Have two admin review flags, something like `admin_code_review` and `admin_content_review`.
- [ ] Content review can still be done for add-ons that have `admin_code_review`, by any content reviewer.
- [ ] Code review can still be done for add-ons that have `admin_content_review`, by any code reviewer.
- [ ] The code review queues only display the `admin_code_review` flag.
- [ ] The content review queues only display the `admin_content_review` flag.
- [ ] Only admin content reviewers can do a content review for an add-on with the `admin_content_review` flag.
- [ ] The same permission is used for admin code reviewers and admin content reviewers (`ReviewerAdminTools:View`, I think). 
- [ ] Admins can unflag add-ons in their respective review pages.

### PR DESCRIPTION
**Pull request by diox**
Fix #6826

### CODE REVIEW

File: src/olympia/devhub/views.py [hunk 1]
```diff
@@ -1333,7 +1334,8 @@ def _submit_upload(request, addon, channel, next_details, next_finish,
         check_validation_override(request, form, addon, version)
         addon_update = {}
```
Comment (eviljeff) [inline src/olympia/devhub/views.py]: we just use this to avoid multiple updates to Addon, but now `admin_review` isn't a property then the only possible update is `status = amo.STATUS_NOMINATED` so we can do it straight away.

File: src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- Make the old field nullable to prepare for its future removal.
+ALTER TABLE `addons` MODIFY COLUMN `adminreview` tinyint(1) NULL;
```
Comment (eviljeff) [inline src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql]: Can you file the follow-up issue now so it's not forgotten.

File: src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+-- Create the new table.
+CREATE TABLE `addons_addonreviewerflags` (
+    `created` datetime(6) NOT NULL,
+    `modified` datetime(6) NOT NULL,
+    `addon_id` integer NOT NULL PRIMARY KEY,
+    `needs_admin_code_review` bool NOT NULL,
+    `needs_admin_content_review` bool NOT NULL
+)
+;
+ALTER TABLE `addons_addonreviewerflags` ADD CONSTRAINT `addon_id_refs_id_7a280313` FOREIGN KEY (`addon_id`) REFERENCES `addons` (`id`);
```
Comment (eviljeff) [inline src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql]: "addon_id_refs_id_7a280313" - a memorable constraint name 😉

File: src/olympia/reviewers/forms.py [hunk 4]
```diff
@@ -130,8 +128,9 @@ def clean_addon_type_ids(self):
 
     def filter_qs(self, qs):
         data = self.cleaned_data
-        if data['admin_review']:
-            qs = qs.filter(admin_review=data['admin_review'])
+        if data['needs_admin_code_review']:
+            qs = qs.filter(
+                needs_admin_code_review=data['needs_admin_code_review'])
```
Comment (eviljeff) [inline src/olympia/reviewers/forms.py]: scope creep, but should we allow filtering by `needs_admin_content_review` too?

File: src/olympia/versions/models.py [hunk 5]
```diff
@@ -493,8 +493,12 @@ def sources_provided(self):
...
 
     @property
-    def admin_review(self):
-        return self.addon.admin_review
+    def needs_admin_code_review(self):
+        return self.addon.needs_admin_code_review
+
+    @property
+    def needs_admin_content_review(self):
+        return self.addon.needs_admin_content_review
```
Comment (eviljeff) [inline src/olympia/versions/models.py]: Can we drop these two properties?  Seems like they are unnecessary shortcuts.

File: src/olympia/devhub/views.py (hunk 1 above)
Comment (diox) [reply src/olympia/devhub/views.py]: Good point, doing that.

File: src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql (hunk 2 above)
Comment (diox) [reply src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql]: Done in #7000

File: src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql (hunk 3 above)
Comment (diox) [reply src/olympia/migrations/989-replace-adminreview-with-addonreviewerflags.sql]: It's generated by django, so it sucks but it's a lot better than our custom ones that make us inconsistent with django syncdb/migrations :)

File: src/olympia/reviewers/forms.py (hunk 4 above)
Comment (diox) [reply src/olympia/reviewers/forms.py]: Maybe in a follow-up, yes. It's unclear what are the exact needs regarding this.

File: src/olympia/versions/models.py (hunk 5 above)
Comment (diox) [reply src/olympia/versions/models.py]: They are necessary to make flags work on the review page itself (`get_flags()` is passed a `Version`)
